ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go

```diff
@@ -287,12 +293,67 @@ func getTaskMetadataForFault(w http.ResponseWriter, agentState state.AgentState,
 		return nil, errResponse
 	}
 
-	// TODO: Check if task is FIS-enabled
-	// TODO: Check if task is using a valid network mode
+	// Check if task is FIS-enabled
+	if !taskMetadata.FaultInjectionEnabled {
+		errResponse := fmt.Sprintf(faultInjectionEnabledError, taskMetadata.TaskARN)
+		responseBody := types.NewNetworkFaultInjectionErrorResponse(errResponse)
+		logger.Error("Error: Task is not fault injection enabled.", logger.Fields{
+			field.RequestType:             requestType,
+			field.TMDSEndpointContainerID: endpointContainerID,
+			field.Response:                responseBody.ToString(),
+			field.TaskARN:                 taskMetadata.TaskARN,
+			field.Error:                   errResponse,
+		})
+		utils.WriteJSONResponse(
+			w,
+			http.StatusBadRequest,
+			responseBody,
+			requestType,
+		)
+		return nil, errors.New(errResponse)
+	}
+
+	if err := validateTaskNetworkConfig(taskMetadata.TaskNetworkConfig); err != nil {
+		code, errResponse := getTaskMetadataErrorResponse(endpointContainerID, requestType, err)
+		responseBody := types.NewNetworkFaultInjectionErrorResponse(fmt.Sprintf("%v", errResponse))
+		logger.Error("Error: Unable to resolve task network config within task metadata", logger.Fields{
+			field.Error:                   errResponse,
+			field.RequestType:             requestType,
+			field.Response:                responseBody.ToString(),
+			field.TMDSEndpointContainerID: endpointContainerID,
+		})
```

Comment:
I think we should add `err` to the log as it's not a part of `errResponse` anymore.

Reply:
good catch